source/common/http/utility.cc

```diff
@@ -562,5 +563,62 @@ void Utility::traversePerFilterConfigGeneric(
   }
 }
 
+std::string Utility::PercentEncoding::encode(absl::string_view value) {
+  for (size_t i = 0; i < value.size(); ++i) {
+    const char& ch = value[i];
+    // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses.
+    if (ch < ' ' || ch >= '~' || ch == '%') {
```

Comment:
This first check is a performance optimization, right? It took me a little while to figure out what was going on here. If so, can you add an explanatory comment about why we are doing it this way?

Reply:
Sure. Added some comments.